### PR TITLE
Don't create null fields if trying to lowercase/uppercase a field tha…

### PR DESCRIPTION
**Pull request by andrewvc**
Rebase based on https://github.com/logstash-plugins/logstash-filter-mutate/pull/66 that adds tests / works with more modern stuff.

### CODE REVIEW

File: lib/logstash/filters/mutate.rb [hunk 1]
```diff
@@ -349,6 +350,7 @@ def lowercase(event)
     #see comments for #uppercase
     @lowercase.each do |field|
       original = event.get(field)
+      next unless original
```
Comment (magnusbaeck) [inline lib/logstash/filters/mutate.rb]: Perhaps use `next if original.nil?` instead so we don't skip non-truthy booleans?

Comment (jsvd) [inline lib/logstash/filters/mutate.rb]: +1

**Comment by jsvd** [comment]
LGTM after addressing Magnus' remark

**Comment by jsvd** [comment]
And I'd suggest adding a test for the boolean case.

File: lib/logstash/filters/mutate.rb (hunk 1 above)
Comment (andrewvc) [inline lib/logstash/filters/mutate.rb]: Good point!

**Pull request closed without merging**